### PR TITLE
`addContender` by signature only

### PR DESCRIPTION
**Pull request by godzillaba**

### CODE REVIEW

File: src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol [hunk 1]
```diff
@@ -215,11 +216,16 @@ contract SecurityCouncilNomineeElectionGovernor is
     ///         during the vetting phase and exclude any contenders which dont meet this criteria.
     /// @dev    Can be called only while a proposal is pending (after proposal created but before voting phase)
     ///         A contender cannot be a member of the opposite cohort.
-    function addContender(uint256 proposalId) external {
```
Comment (DZGoldman) [inline src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol:218]: maybe let's keep the old addContender too and mark it as deprecated / revert

**Review by DZGoldman — CHANGES_REQUESTED** [review]

File: src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol (hunk 1 above)
Comment (godzillaba) [reply src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol:218]: like this? 72c7c399a121ddb69db87635c3aefca6dab19a15

**Review by DZGoldman — APPROVED** [review]

File: src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol [hunk 2]
```diff
@@ -475,6 +491,11 @@ contract SecurityCouncilNomineeElectionGovernor is
         );
     }
 
+    /// @notice Deprecated, use `addContender(uint256 proposalId, bytes calldata signature)` instead
+    function addContender(uint256) external pure {
```
Comment (fredlacs) [inline src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol:495]: we can have 2 optional entrypoints.
addContederImpl that is called with prop id and sender
the signature entrypoint and msg.sender entrypoint both call it.

keeps the code backward compatible

**Review by fredlacs — CHANGES_REQUESTED** [review]

File: src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol (hunk 2 above)
Comment (DZGoldman) [reply src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol:495]: Depends on the motivation behind the change; if we want to guarantee that the sender can produce a signature, then we may want to disallow the old addContender (allows for smart contract wallet, address alias, etc. that can't necessarily sign on all governed chains). 

if the motivation is just to add a new feature, then we keep it

Comment (fredlacs) [reply src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol:495]: thats true, lets not keep it to ensure that by not keeping it backwards compatible.
would be good to add comments alongside the code that explain the above

Comment (DZGoldman) [reply src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol:495]: > thats true, lets not keep it to ensure that by not keeping it backwards compatible.

so just to be sure – keep the logic as it currently is on this branch (`addContender(uint256)` method still exists but just reverts with a deprecation error) but add more natspec comments explaining the rationale (restrict contender-adding to only an address that can produce a sig)

**Review by DZGoldman — APPROVED** [review]

**Comment by DZGoldman** [comment]
(feel free to merge if / when ready)

**Pull request merged**